src/Microsoft.AspNet.Authentication/DataHandler/TicketSerializer.cs

```diff
@@ -36,98 +39,177 @@ public virtual AuthenticationTicket Deserialize(byte[] data)
             }
         }
 
-        public static void Write([NotNull] BinaryWriter writer, [NotNull] AuthenticationTicket model)
+        public virtual void Write([NotNull] BinaryWriter writer, [NotNull] AuthenticationTicket ticket)
```

Comment:
I haven't changed the visibility modifier of these methods, but this is something we can consider.

That said, note that `PropertiesSerializer.Write/Read` is internally used by the Twitter middleware.
If you want to make everything protected internal, we'll have to add a `InternalsVisibleToAttribute` for both the test project and the Twitter provider.